server/controllers/connectedAccounts.js

```diff
@@ -116,12 +118,24 @@ export const disconnect = async (req, res) => {
       throw new errors.Unauthorized('You are either logged out or not authorized to disconnect this account');
     }
 
-    const account = await models.ConnectedAccount.findOne({
-      where: { service, CollectiveId },
-    });
+    if (service === 'transferwise') {
+      const collective = await models.Collective.findByPk(CollectiveId);
+      const connectedAccounts = await models.ConnectedAccount.findAll({
+        where: { service, CollectiveId },
+      });
+      if (collective.settings?.transferwise?.isolateUsers) {
+        const account = connectedAccounts.find(ca => ca.CreatedByUserId === remoteUser.id);
+        assert(account, 'No connected account found for this user');
+        await account.destroy();
+      }
+    } else {
+      const account = await models.ConnectedAccount.findOne({
+        where: { service, CollectiveId },
+      });
```

Comment:
That means peeps won't be able to disconnect their accounts if they have the `isolateUsers` flag; shouldn't we do:

```suggestion
    const collective = await models.Collective.findByPk(CollectiveId);
    if (service === 'transferwise' && collective.settings?.transferwise?.isolateUsers) {
      const connectedAccounts = await models.ConnectedAccount.findAll({
        where: { service, CollectiveId },
      });
      const account = connectedAccounts.find(ca => ca.CreatedByUserId === remoteUser.id);
      assert(account, 'No connected account found for this user');
      await account.destroy();
    } else {
      const account = await models.ConnectedAccount.findOne({
        where: { service, CollectiveId },
      });
```

Reply:
Nice catch, Ben!